File: samples/snippets/cpp/VS_Snippets_Winforms/Windows.Forms.Control Properties3/CPP/aboutdialog.cpp


#using <System.Windows.Forms.dll>
#using <System.dll>
#using <System.Drawing.dll>

using namespace System;
using namespace System::Drawing;
using namespace System::ComponentModel;
using namespace System::Windows::Forms;
using namespace System::Reflection;
using namespace System::Runtime::CompilerServices;

[assembly:AssemblyCompany("Microsoft")];
[assembly:AssemblyProduct("Test")];
[assembly:AssemblyVersion("1.0.*")];
namespace CodeExamples
{
   public ref class AboutDialog: public System::Windows::Forms::Form
   {
   private:
      System::Windows::Forms::Button^ buttonOK;
      System::Windows::Forms::Label ^ labelVersionInfo;
      System::ComponentModel::Container^ components;

   public:
      AboutDialog()
      {
         InitializeComponent();
      }

   protected:
   ~AboutDialog()
   {
      if ( components != nullptr )
      {
         delete components;
      }
   }

   private:
      void InitializeComponent()
      {
         this->labelVersionInfo = gcnew System::Windows::Forms::Label;
         this->buttonOK = gcnew System::Windows::Forms::Button;
         this->SuspendLayout();

         //
         // labelVersionInfo
         //
         this->labelVersionInfo->Location = System::Drawing::Point( 8, 8 );
         this->labelVersionInfo->Name = "labelVersionInfo";
         this->labelVersionInfo->Size = System::Drawing::Size( 360, 23 );
         this->labelVersionInfo->TabIndex = 0;
         this->labelVersionInfo->Text = "label1";

         //
         // buttonOK
         //
         this->buttonOK->Location = System::Drawing::Point( 296, 64 );
         this->buttonOK->Name = "buttonOK";
         this->buttonOK->TabIndex = 3;
         this->buttonOK->Text = "&OK";

         //
         // AboutDialog
         //
         this->ClientSize = System::Drawing::Size( 392, 109 );
         array<System::Windows::Forms::Control^>^temp0 = {this->buttonOK,this->labelVersionInfo};
         this->Controls->AddRange( temp0 );
         this->Name = "AboutDialog";
         this->Text = "About";
         this->Load += gcnew System::EventHandler( this, &AboutDialog::AboutDialog_Load );
         this->ResumeLayout( false );
      }

      // <snippet1>
      void AboutDialog_Load( Object^ /*sender*/, EventArgs^ /*e*/ )
      {
         // Display the application information in the label.
         this->labelVersionInfo->Text = String::Format(  "{0} {1} Version: {2}", this->CompanyName, this->ProductName, this->ProductVersion );
      }
      // </snippet1>
   };
}

[STAThread]
int main()
{
   Application::Run( gcnew CodeExamples::AboutDialog );
}

File: samples/snippets/visualbasic/VS_Snippets_Winforms/Windows.Forms.Control Properties3/VB/aboutdialog.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Reflection
Imports Microsoft.VisualBasic
Imports System.Runtime.CompilerServices

<Assembly: AssemblyCompany("Microsoft")>
<Assembly: AssemblyProduct("Test")>
<Assembly: AssemblyVersion("1.0.*")>

Namespace CodeExamples
   Public Class AboutDialog
      Inherits System.Windows.Forms.Form
      Private buttonOK As System.Windows.Forms.Button
      Private labelVersionInfo As System.Windows.Forms.Label
      Private components As System.ComponentModel.Container = Nothing
      
      
      Public Sub New()
         InitializeComponent()
      End Sub 'New
      
      Private Sub InitializeComponent()
         Me.labelVersionInfo = New System.Windows.Forms.Label()
         Me.buttonOK = New System.Windows.Forms.Button()
         Me.SuspendLayout()
         ' 
         ' labelVersionInfo
         ' 
         Me.labelVersionInfo.Location = New System.Drawing.Point(8, 8)
         Me.labelVersionInfo.Name = "labelVersionInfo"
         Me.labelVersionInfo.Size = New System.Drawing.Size(360, 23)
         Me.labelVersionInfo.TabIndex = 0
         Me.labelVersionInfo.Text = "label1"
         ' 
         ' buttonOK
         ' 
         Me.buttonOK.Location = New System.Drawing.Point(296, 64)
         Me.buttonOK.Name = "buttonOK"
         Me.buttonOK.TabIndex = 3
         Me.buttonOK.Text = "&OK"
         ' 
         ' AboutDialog
         ' 
         Me.ClientSize = New System.Drawing.Size(392, 109)
         Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.buttonOK, Me.labelVersionInfo})
         Me.Name = "AboutDialog"
         Me.Text = "About"
         Me.ResumeLayout(False)
      End Sub 'InitializeComponent
       
     Shared Sub Main()
         Application.Run(New AboutDialog())
      End Sub 'Main
      
      
' <snippet1>
Private Sub AboutDialog_Load(sender As Object, e As EventArgs) Handles MyBase.Load
   ' Display the application information in the label.
   Me.labelVersionInfo.Text = _
      Me.CompanyName + "  " + _
      Me.ProductName + "  Version: " + _
      Me.ProductVersion
   End Sub
' </snippet1>


   End Class 'AboutDialog
End Namespace 'ControlMembers2
